engine/src/main/java/org/hibernate/validator/messageinterpolation/AbstractMessageInterpolator.java

```diff
@@ -356,10 +357,12 @@ private String interpolateMessage(String message, Context context, Locale locale
 	}
 
 	private String replaceEscapedLiterals(String resolvedMessage) {
-		resolvedMessage = LEFT_BRACE.matcher( resolvedMessage ).replaceAll( "{" );
-		resolvedMessage = RIGHT_BRACE.matcher( resolvedMessage ).replaceAll( "}" );
-		resolvedMessage = SLASH.matcher( resolvedMessage ).replaceAll( Matcher.quoteReplacement( "\\" ) );
-		resolvedMessage = DOLLAR.matcher( resolvedMessage ).replaceAll( Matcher.quoteReplacement( "$" ) );
+		if ( resolvedMessage.indexOf( '\\' ) > -1 ) {
```

Comment:
BTW, I think we could use the same optimization before triggering the interpolation mechanisms: we don't need to interpolate if the string does not contain '{' - in fact, we could even skip the EL interpolation if the string does not contain "${" but I'm not sure it's worth the additional call to `contains()`.

It's probably worth a quick benchmark to be sure the additional `indexOf()` does not slow us down.